src/Assets/Asset.php

```diff
@@ -32,11 +32,15 @@ public function syncOriginal()
 
     public static function fromModel(Model $model)
     {
-        return (new static)
+        $asset = (new static)
             ->container($model->container)
             ->path(Str::replace('//', '/', $model->folder.'/'.$model->basename))
             ->hydrateMeta($model->meta)
             ->syncOriginal();
+
+        Blink::put('asset-meta-'.$asset->id(), $model->meta);
```

Comment:
```suggestion
        Blink::put('eloquent-asset-'.$asset->id(), $model->meta);
```

It works with the above change (used a different key).

You could also use `Blink::put('eloquent-asset-meta-exists-'.$asset->id(), true)` and remove the additional `Blink::has('eloquent-asset-'.$this->id())` in metaExists

Reply:
```suggestion
        Blink::put($asset->metaCacheKey(), $model->meta);
```

`metaCacheKey` is a public method of Assets, so you could use this in a unified way:
https://github.com/statamic/cms/blob/eba307923eb5fe9ce7b0954bf6ceecc0357be8c1/src/Assets/Asset.php#L324-L327

Reply:
Yeah, the reduction still works with the latest changes.